src/pytest_dbfixtures/factories/mysql.py

```diff
@@ -104,7 +105,7 @@ def mysql_proc_fixture(request):
         mysql_host = host or config.mysql.host
         mysql_params = params or config.mysql.params
 
-        tmpdir = path(gettempdir())
+        tmpdir = path(mkdtemp(prefix="pytest-mysql-"))
```

Comment:
mysqld will create files directly in `--tmpdir`, so it's not suitable to be $TMPDIR directly. Alternatively we could create another tmpdir inside the tmpdir returned by `gettempdir`, there wouldn't be much difference. I chose to put all files into one directory to keep $TMPDIR little more tidy.

Btw, even `mkdtemp` is affected by $TMPDIR.